demo/image_creation.sh

```diff
@@ -1,125 +1,162 @@
 echo "Done"
 
 echo "*********************************************************************"
-echo "Creating vm $vm with user $admin_username in $location"
+echo "Creating vm $VM_NAME with user $ADMIN_USERNAME in $LOCATION"
 echo "*********************************************************************"
-az vm create -g $rg -n $vm --image $base_image --admin-username $admin_username --ssh-key-value @${ssh_key_path} --boot-diagnostics-storage $storage --size Standard_D2ds_v5 --accelerated-network true --nic-delete-option Delete --os-disk-delete-option Delete --custom-data $temp_dir/customdata.yml
+az vm create -g "$RG" -n "$VM_NAME" --image "$BASE_IMAGE" --admin-username "$ADMIN_USERNAME" --ssh-key-value @${SSH_KEY_PATH} --boot-diagnostics-storage "$STORAGE_ACCOUNT" --size "${VM_SIZE}" --accelerated-network true --nic-delete-option Delete --os-disk-delete-option Delete --custom-data "$TEMP_DIR"/customdata.yml --security-type "$SECURITY_TYPE"
 echo "vm finished deployment, waiting for image configuration to finish"
 
-deadline=$(date -u -d '15 minutes' '+%s')
-found=0
-while [[ $(date '+%s') < $deadline ]]
+DEADLINE=$(date -u -d '15 minutes' '+%s')
+FOUND=0
+while [[ $(date '+%s') < $DEADLINE ]]
 do
-    power_state=$(az vm get-instance-view -g $rg -n $vm | jq '.instanceView.statuses[] | select(.code | contains("PowerState")) | .code' -r)
-    if [[ "$power_state" == *"stopped"* ]]
+    POWER_STATE=$(az vm get-instance-view -g "$RG" -n "$VM_NAME" | jq '.instanceView.statuses[] | select(.code | contains("PowerState")) | .code' -r)
+    if [[ "$POWER_STATE" == *"stopped"* ]]
     then
-        log=$(az vm boot-diagnostics get-boot-log -g $rg -n $vm)
-        if [[ -n "$log" && "$log" == *"SIGTOOL_END"* ]]
+        LOG=$(az vm boot-diagnostics get-boot-log -g "$RG" -n "$VM_NAME")
+        if [[ -n "$LOG" && "$LOG" == *"SIGTOOL_END"* ]]
         then
             echo "vm configured successfully"
-            found=1
+            FOUND=1
             break
         fi
     fi
     sleep 15
 done
 
-if [[ $found -eq 0 ]]
+if [[ $FOUND -eq 0 ]]
 then
     echo "vm failed to configure in 15 minutes - abort"
     exit 1
 fi
 
-image=image-$epoch
+IMAGE_NAME=image-$EPOCH
+echo "*********************************************************************"
+echo "Capturing OsDisk snapshot of vm $VM_NAME with image $IMAGE_NAME"
+echo "*********************************************************************"
+TARGET_DISK=$(az disk show --ids "$(az vm show -g "$RG" -n "$VM_NAME" | jq .storageProfile.osDisk.managedDisk.id -r)" | jq .name -r)
+az snapshot create -g "$RG" -n "$VM_NAME"-snapshot --source "$TARGET_DISK"
+
+IMAGE_VERSION=$(date '+%Y.%m%d.%H%M%S')
+GALLERY_NAME=testgalleryazinit
+GALLERY_DEFINITION=testgalleryazinitdef
+GALLERY_RG=testgalleryazinitrg
+SNAPSHOT_ID=$(az snapshot show -n "$VM_NAME"-snapshot -g "$RG" --query id --output tsv)
+
+# If BASE_IMAGE == canonical:0001-com-ubuntu-server-jammy:22_04-lts-gen2:latest,
```

Comment:
remove these commented lines

Reply:
Actually the comments are necessary descriptions including examples, not accidentally left.

Will instead change like below for better descriptions:
```
# If BASE_IMAGE is set to canonical:0001-com-ubuntu-server-jammy:22_04-lts-gen2:latest, then
# GALLERY_PUBLISHER becomes "canonical"
# GALLERY_OFFER becomes "0001-com-ubuntu-server-jammy"
# GALLERY_SKU becomes "22_04-lts-gen2"
```

Reply:
Done